### DISCUSSION TITLE
Can we know from the sketch which is the renderer being used? or at least if it is P3D?

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#540] [Ideas]
There was this [old discussion](https://forum.processing.org/one/topic/how-to-tell-if-renderer-is-3d.html) that pointed to the `g` methods `g.is2D()` and `g.is3D()`. 

This would be useful for me because my naive Slider code needs to call `camera()` to properly reset the view when P3D is in use (it tries to emulate some kind of HUD), but not otherwise, as it shows a warning if you are not using P3D...

Well maybe I can just use ` get_graphics()._instance.is3D()`...

**Comment by hx2A** [comment]
Right, you can call `py5.get_graphics()._instance.is3D()`. I could make that into a real py5 method in `Py5Graphics` so you don't need the `._instance` part. It's an easy thing to add.

Keep in mind that currently the only 3D renderer offered by Processing is the P3D one but in the future there may be other 3D renderers.

If you want the name of the actual renderer, there's also another internal function you can access with `py5.get_current_sketch()._instance.getRendererName()`, which will return a Java string like `JAVA2D`, `P3D`, etc. or `py5.get_current_sketch()._instance.sketchRenderer()` which will return string for the renderer class.

**Reply by villares** [reply]
cool... looking forward to the day that wonderful hidden **** renderer comes out ;-)

**Reply by hx2A** [reply]
http://dev.py5coding.org/reference/py5graphics_is3d.html

It was easy and I had time to do this while waiting for something else to finish. It will be in the next release.

**Reply by hx2A** [reply]
I was going to make it a property so you wouldn't need the `()` but then I noticed Py5Shape has a method with the same name, so I thought it would be best to make them consistent.
